boxsdk/session/session.py

```diff
@@ -541,6 +556,10 @@ def _get_retry_request_callable(
             self._renew_session(request.access_token)
             request.auto_session_renewal = False
             return self._send_request
+
+        if request.method == 'GET' and network_response and network_response.ok and not self._validate_json_response(network_response, request):
```

Comment:
You put this piece into AuthorizedSession class. I guess the better place would be to add it to parent class Session to overriden method, because this thing can happen both dor authorized and unauthorized session